### PR TITLE
feat(auth): system api keys ui

### LINKED ISSUE #4269: [auth][ui] system api keys

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #4269 

Screenshot 2024-08-19 at 2 54 11 PM

### CODE REVIEW

File: app/src/pages/settings/APIKeysCard.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      <div
+        css={css`
+          padding: var(--ac-global-dimension-size-200);
+          .ac-field {
+            width: 100%;
+          }
+        `}
+      >
+        <Flex direction="row" gap="size-100" alignItems="end">
+          <TextField label="API Key" isReadOnly value={jwt} minWidth="100%" />
```
Comment (Parker-Stafford) [inline app/src/pages/settings/APIKeysCard.tsx:104]: do we want to obscure it even though it's only displayed once - **** with copy to clipboard - change messaging a bit: "The API key will only be copyable once below..."

File: app/src/pages/settings/CreateSystemAPIKeyDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        </Alert>
+      )}
+      <Form onSubmit={handleSubmit(onSubmit)}>
+        <View padding="size-200">
+          <Controller
+            name="name"
+            control={control}
+            rules={{
+              required: "System key name is required",
+            }}
```
Comment (Parker-Stafford) [inline app/src/pages/settings/CreateSystemAPIKeyDialog.tsx:90]: for my product level understanding - there can be multiple system keys, and they can all have different names, correct?

File: app/src/pages/settings/APIKeysCard.tsx (hunk 1 above)
Comment (mikeldking) [reply app/src/pages/settings/APIKeysCard.tsx:104]: typically these are displayed once. it's similar to how github does thsee

File: app/src/pages/settings/CreateSystemAPIKeyDialog.tsx (hunk 2 above)
Comment (mikeldking) [reply app/src/pages/settings/CreateSystemAPIKeyDialog.tsx:90]: Yup, and they can ultimately have the same name if they want

**Review by Parker-Stafford — APPROVED** [review]

**Pull request merged**